Ursa spack-stack/2.0.0 bacio/2.6.0 INC and LIB variables point at non-existent paths

## Description

While `module show bacio/2.6.0` returns
```
setenv("BACIO_LIB4", "/contrib/spack-stack/spack-stack-2.0.0/envs/ue-oneapi-2025.2.1/install/intel-oneapi-compilers/2025.2.1/bacio-2.6.0-2ecrwd4/lib/libbacio_4.a")
setenv("BACIO_LIB8", "/contrib/spack-stack/spack-stack-2.0.0/envs/ue-oneapi-2025.2.1/install/intel-oneapi-compilers/2025.2.1/bacio-2.6.0-2ecrwd4/lib/libbacio_8.a")
```
directory `/contrib/spack-stack/spack-stack-2.0.0/envs/ue-oneapi-2025.2.1/install/intel-oneapi-compilers/2025.2.1/bacio-2.6.0-2ecrwd4/lib` does not exist. 

Instead we have `/contrib/spack-stack/spack-stack-2.0.0/envs/ue-oneapi-2025.2.1/install/intel-oneapi-compilers/2025.2.1/bacio-2.6.0-2ecrwd4/lib64` which contains
```
drwxr-sr-x 3 role.epic epic  2048 Jan  6 21:29 cmake
-rw-r--r-- 1 role.epic epic 27870 Jan  6 21:29 libbacio.a
```

The same comment applies to `BACIO_INC4` and `BACIO_INC8`.  Neither of directories to which bacio/2.6.0 points exist.

## To Reproduce

Execute the following on Ursa
```
module purge
module use /contrib/spack-stack/spack-stack-2.0.0/envs/ue-oneapi-2025.2.1/modules/Core
module load stack-intel-oneapi-compilers/2025.2.1
module load bacio/2.6.0
module show bacio/2.6.0
```

The paths for `BACIO_INC4`, `BACIO_INC8`, `BACIO_LIB4`, and `BACIO_LIB8` do not exist.

## Expected behavior

The paths for `BACIO_INC4`, `BACIO_INC8`, `BACIO_LIB4`, and `BACIO_LIB8` should exist and be populated with the appropriate files.

## System, compiler, code, ...

stack-intel-oneapi-compilers/2025.2.1

## Additional context

Encountered this issue while attempting to build [NOAA-EMC/GSI](https://github.com/NOAA-EMC/GSI) on Ursa using spack-stack/2.0.0.  

GSI is built with [`bacio_4`](https://github.com/NOAA-EMC/GSI/blob/5f65a97fc00c9d5ad060402e4d47390b72999c13/src/gsi/CMakeLists.txt#L150C1-L150C58).  The build failed because bacio_4 could not be found.

Thank you @climbfuji .  I replaced `bacio_4` with `bacio`.  The GSI build got a bit further but died with
```
CMake Error at src/gsi/CMakeLists.txt:158 (target_link_libraries):
  Target "gsi_fortran_obj" links to:

    crtm::crtm

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.
```

We currently build GSI with crtm/2.4.0.1 when using spack-stack/1.9.2.  crtm/2.4.0.1 needs stack-oneapi/2024.2.1 but when we move to spack-stack/2.0.0 we use stack-intel-oneapi-compilers/2025.2.1.  

I tried crtm/3.1.2 and got the above build error.

> Thank you [@climbfuji](https://github.com/climbfuji) . I replaced `bacio_4` with `bacio`. The GSI build got a bit further but died with
> 
> ```
> CMake Error at src/gsi/CMakeLists.txt:158 (target_link_libraries):
>   Target "gsi_fortran_obj" links to:
> 
>     crtm::crtm
> 
>   but the target was not found.  Possible reasons include:
> 
>     * There is a typo in the target name.
>     * A find_package call is missing for an IMPORTED target.
>     * An ALIAS target is missing.
> ```
> 
> We currently build GSI with crtm/2.4.0.1 when using spack-stack/1.9.2. crtm/2.4.0.1 needs stack-oneapi/2024.2.1 but when we move to spack-stack/2.0.0 we use stack-intel-oneapi-compilers/2025.2.1.
> 
> I tried crtm/3.1.2 and got the above build error.

@DavidHuber-NOAA I suppose you've seen this before?

--- a/configs/common/modules_lmod.yaml
+++ b/configs/common/modules_lmod.yaml
@@ -162,10 +162,8 @@ modules:
       bacio:
         environment:
           set:
-            'BACIO_INC4': '{prefix}/include_4'
-            'BACIO_INC8': '{prefix}/include_8'
-            'BACIO_LIB4': '{prefix}/lib/libbacio_4.a'
-            'BACIO_LIB8': '{prefix}/lib/libbacio_8.a'
+            'BACIO_INC': '{prefix}/include'
+            'BACIO_LIB': '{prefix}/lib/libbacio.a'
       bufr:
         environment:
           set:

--- a/configs/common/modules_lmod.yaml
+++ b/configs/common/modules_lmod.yaml
@@ -163,7 +163,7 @@ modules:
         environment:
           set:
             'BACIO_INC': '{prefix}/include'
-            'BACIO_LIB': '{prefix}/lib/libbacio.a'
+            'BACIO_LIB': '{prefix}/lib64/libbacio.a'
       bufr:
         environment:
           set:

--- a/configs/common/modules_lmod.yaml
+++ b/configs/common/modules_lmod.yaml
@@ -172,7 +172,7 @@ modules:
         environment:
           set:
             'CRTM_INC': '{prefix}/include'
-            'CRTM_LIB': '{prefix}/lib/libcrtm.a'
+            'CRTM_LIB': '{prefix}/lib/libcrtm.so'
       g2tmpl:
         environment:
           set:

--- a/configs/common/modules_tcl.yaml
+++ b/configs/common/modules_tcl.yaml
@@ -181,10 +181,8 @@ modules:
       bacio:
         environment:
           set:
-            'BACIO_INC4': '{prefix}/include_4'
-            'BACIO_INC8': '{prefix}/include_8'
-            'BACIO_LIB4': '{prefix}/lib/libbacio_4.a'
-            'BACIO_LIB8': '{prefix}/lib/libbacio_8.a'
+            'BACIO_INC': '{prefix}/include'
+            'BACIO_LIB': '{prefix}/lib/libbacio.a'
       bufr:
         environment:
           set:

--- a/configs/common/modules_tcl.yaml
+++ b/configs/common/modules_tcl.yaml
@@ -182,7 +182,7 @@ modules:
         environment:
           set:
             'BACIO_INC': '{prefix}/include'
-            'BACIO_LIB': '{prefix}/lib/libbacio.a'
+            'BACIO_LIB': '{prefix}/lib64/libbacio.a'
       bufr:
         environment:
           set:

--- a/configs/common/modules_tcl.yaml
+++ b/configs/common/modules_tcl.yaml
@@ -191,7 +191,7 @@ modules:
         environment:
           set:
             'CRTM_INC': '{prefix}/include'
-            'CRTM_LIB': '{prefix}/lib/libcrtm.a'
+            'CRTM_LIB': '{prefix}/lib/libcrtm.so'
       g2tmpl:
         environment:
           set:

